### PR TITLE
Wrapper: look through deprecated functions when evaluating radspec

### LINKED ISSUE #340: Make Radspec descriptions work on breaking upgrades
Following the work done on the aragonCLI (https://github.com/aragon/aragon-cli/issues/450)

Been able to handle deprecated functions ABIs on the generated `artifact.json`.

### PR DESCRIPTION
**Pull request by 0xGabi**
**Changes**

Update `findAppMethodFromData` function to find method on new `deprecated` value of artifact.json if not found. 

Closes #340 

TODO:

- [x] Test with Finance app
- [x] Test with apps with no deprecated functions

### CODE REVIEW

**Comment by sohkai** [comment]
@0xGabi I've pushed a small refactor of the code to make it a bit more robust in 8fe17bd. It relies on https://github.com/aragon/aragon-cli/pull/426/files#r306909084 though :).

File: packages/aragon-wrapper/src/utils/apps.js [hunk 1]
```diff
@@ -12,13 +12,28 @@ export const apmAppId = appName => namehash(`${appName}.aragonpm.eth`)
...
+  if (Array.isArray(functions)) {
+    method = functions.find(
+      method => keccak256(method.sig).substring(0, 8) === methodId
+    )
+  }
+
+  // Couldn't find it in current functions, try on deprecated versions
+  if (!method && Array.isArray(deprecatedFunctions)) {
+    // Flatten all the deprecated functions
+    const allDeprecatedFunctions = [].concat(...Object.values(deprecatedFunctions))
```
Comment (0xGabi) [inline packages/aragon-wrapper/src/utils/apps.js]: I was wondering how to do something like this 👏

File: packages/aragon-wrapper/src/utils/apps.js [hunk 2]
```diff
@@ -12,13 +12,28 @@ export const apmAppId = appName => namehash(`${appName}.aragonpm.eth`)
...
+
+  // First try to find the method in the current functions
+  if (Array.isArray(functions)) {
+    method = functions.find(
+      method => keccak256(method.sig).substring(0, 8) === methodId
+    )
+  }
+
+  // Couldn't find it in current functions, try on deprecated versions
+  if (!method && Array.isArray(deprecatedFunctions)) {
```
Comment (0xGabi) [inline packages/aragon-wrapper/src/utils/apps.js]: I think `Array.isArray` here will not be useful as the `deprecatedFunctions` structure is:
```
"deprecatedFunctions": {
    "5.0.0": [
        "f1": {...}
        "f2": {...}
    ],
    "3.0.0": [
       "f1": {...}
    ]
}
```

**Comment by sohkai** [comment]
@0xGabi I've pushed 81e9fa6 as a small refactor and to fix an issue with `.every()` being true if the array length is 0.

Will try to deploy Finance with aragonCLI v6 onto staging to test this :+1:.

**Comment by sohkai** [comment]
@0xGabi Finally got around to re-testing this branch and fixing it up a bit, as it seemed like a better solution to the radspec problem we're having on Finance with deprecated functions.

It looks good now and will merge and re-release :).

**Pull request merged**